src/components/Modals/NewCourse/NewCourseModal.vue

```diff
@@ -136,9 +133,13 @@ export default defineComponent({
       if (this.selectedCourse == null) return;
       this.$emit('add-course', this.selectedCourse, {
         // Only exclude the selected requirement from opt-out.
-        optOut: this.relatedRequirements
-          .filter(it => it.id !== this.selectedRequirementID)
-          .map(it => it.id),
+        optOut: getRelatedRequirementIdsForCourseOptOut(
+          this.selectedCourse.crseId,
+          this.selectedRequirementID,
+          store.state.groupedRequirementFulfillmentReport,
+          store.state.toggleableRequirementChoices,
+          store.state.userRequirementsMap
+        ),
```

Comment:
This fixes the following issue:
College: ENG
Major: CS
Minor: Human Development

Add HD 1150 as a course through the add course modal / semester. Due to the limitation of the add modal, the user cannot choose more than one non-double countable requirement. However, after adding the course, the dangerous graph should have all edges not in the constraint violation (eg. if you choose Liberal Studies, both minor requirements should be connected in the dangerous graph).

Reply:
Tested this and confirmed what you are saying is true! 

This isn't ideal on the front-end, as it means that when users choose one of the minor options in the constraint instead of liberal studies, they are still getting liberal studies set to the course. However the new modal will make this flow more clear right?

Reply:
Same issue when adding `CS 1110` where adding it to Computing also adds it to Introductory Programming

Reply:
> This isn't ideal on the front-end, as it means that when users choose one of the minor options in the constraint instead of liberal studies, they are still getting liberal studies set to the course.

I think this is similar to what would happen in the new add flow. There are no conflicts, so we might as well add it. I believe if you choose a non-liberal studies and non-minor requirement, liberal studies would be no longer connected.

Reply:
I modified the function so that it opted out of all `requirementsThatDoNotAllowDoubleCounting`. So, there should be no constraint violations resulting from the minor requirements.

It still automatically selects the liberal studies requirement if it doesn't cause a constraint violation, which is a slight disconnect from the UI. I don't think this can be fixed until the new add modal.